webapp/channels/src/components/color_input.test.tsx

```diff
@@ -13,80 +13,195 @@ describe('components/ColorInput', () => {
-        const wrapper = shallow(
-            <ColorInput {...baseProps}/>,
-        );
+    test('should close color picker when clicking on the picker container', () => {
+        render(<ColorInput {...baseProps}/>);
 
-        wrapper.find('.input-group-addon').simulate('click');
-        wrapper.find('.color-popover').simulate('click');
+        const colorButton = screen.getByTestId('color-picker-button');
+        fireEvent.click(colorButton);
 
-        expect(wrapper).toMatchSnapshot();
+        const pickerModal = screen.getByTestId('color-picker-modal');
+        expect(pickerModal).toBeInTheDocument();
+
+        fireEvent.click(pickerModal);
+
+        // The picker should remain open when clicking on the picker itself
+        expect(screen.getByTestId('color-picker-modal')).toBeInTheDocument();
     });
 
-    test('should have match state on togglePicker', () => {
-        const wrapper = shallow(
-            <ColorInput {...baseProps}/>,
+    test('should handle input focus, change, and blur correctly', async () => {
+        const {rerender} = render(<ColorInput {...baseProps}/>);
+
+        const colorInput = screen.getByTestId('color-inputColorValue');
+        const colorIcon = screen.getByTestId('color-icon');
+
+        // Focus the input
+        fireEvent.focus(colorInput);
+
+        // Clear the input and type new value
+        fireEvent.change(colorInput, {target: {value: '#abc'}});
+
+        // Verify the onChange was called with normalized color
+        expect(baseProps.onChange).toHaveBeenCalledWith('#aabbcc');
+        expect(colorInput).toHaveValue('#abc');
+        expect(colorIcon).toHaveStyle('background-color: #abc');
+
+        // Simulate blur by clicking outside
+        fireEvent.blur(colorInput);
+
+        // Re-render with the updated value from onChange
+        rerender(
+            <ColorInput
+                {...baseProps}
+                value='#aabbcc'
+            />,
         );
 
-        wrapper.setState({isOpened: true});
+        await waitFor(() => {
+            expect(screen.getByDisplayValue('#aabbcc')).toBeInTheDocument();
+            expect(colorIcon).toHaveStyle('background-color: #aabbcc');
+        });
+    });
+
+    test('should handle keyboard interactions to open picker', () => {
+        render(<ColorInput {...baseProps}/>);
+
+        const colorInput = screen.getByTestId('color-inputColorValue');
 
-        wrapper.find('.input-group-addon').simulate('click');
-        expect(wrapper.state('isOpened')).toEqual(false);
+        // Focus input and press Enter
+        fireEvent.focus(colorInput);
+        fireEvent.keyDown(colorInput, {key: 'Enter', code: 'Enter'});
 
-        wrapper.find('.input-group-addon').simulate('click');
-        expect(wrapper.state('isOpened')).toEqual(true);
+        expect(screen.getByTestId('color-picker-modal')).toBeInTheDocument();
     });
 
-    test('should keep what the user types in the textbox until blur', () => {
```

Comment:
It seems like we might be losing this test case. I'm not sure if any of the new cases cover this